src/main/java/com/aws/greengrass/disk/spool/DiskSpool.java

```diff
@@ -76,8 +79,8 @@
     public void add(long id, SpoolMessage message) throws IOException {
         try {
             dao.insertSpoolMessage(message);
-            logger.atTrace().kv("MessageId", id).log("Added message to Disk Spooler");
-        } catch (SQLException e) {
+            logger.atTrace().kv(KV_MESSAGE_ID, id).log("Added message to Disk Spooler");
+        } catch (SQLException | InterruptedException e) { // TODO support InterruptedException in interface
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

**Remember**: Do this only for InterruptedException. Doing this for other types of exceptions may give rise to subtle bugs.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)